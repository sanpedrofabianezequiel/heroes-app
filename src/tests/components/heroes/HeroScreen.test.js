import React from 'react';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { MemoryRouter, Route, Router } from 'react-router';
import { HeroScreen } from '../../../../src/components/heroes/HeroScreen';

describe('Prueabs en <HeroScreen/>',()=>{   

    test('debe de mostrar el componente redirect si no hay argumentos en el URL',()=>{
   
        const historyMock = {
            length :10,
            push:jest.fn(),
            goBack:jest.fn()
        }
   
        const wrapper = mount(
            <MemoryRouter initialEntries={['/heroe']}>
                <HeroScreen  history={historyMock}/>
            </MemoryRouter>
        );
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('Redirect').exists()).toBe(true);

    });

    test('debe de mostrar un heroe si el parametro existe y se encuentra',()=>{

        const wrapper = mount(
            <MemoryRouter initialEntries={['/hero/marvel-spider']}>
                <Route path="/hero/:heroeId" component={HeroScreen} />
            </MemoryRouter>
        );
        expect(wrapper.find('.row').exists()).toBe(true);
    });

    test('debe de regresar a la pantalla anterior con PUSH',()=>{
        const historyMock ={
            length:1,
            push:jest.fn(),
            goBack:jest.fn()
        }
        const wrapper = mount(
            <MemoryRouter initialEntries={['/hero/marvel-spider']}>
                <Route path="/hero/:heroeId" component={()=><HeroScreen history={historyMock} />} />
            </MemoryRouter>
        );
        wrapper.find('button').prop('onClick')();

        expect(historyMock.push).toHaveBeenCalledWith('/');
        expect(historyMock.goBack).not.toHaveBeenCalled();
    });

    test('debe de regresar a la pantalla anterior GOBACK',()=>{
        const historyMock ={
            length:10,
            push:jest.fn(),
            goBack:jest.fn()
        }
        const wrapper = mount(
            <MemoryRouter initialEntries={['/hero/marvel-spider']}>
                <Route path="/hero/:heroeId" component={()=><HeroScreen history={historyMock} />} />
            </MemoryRouter>
        );
        wrapper.find('button').prop('onClick')();

        expect(historyMock.push).not.toHaveBeenCalled();
        expect(historyMock.goBack).toHaveBeenCalled();
    });

    test('debe de llamar el REDIRECT si el hero no existe',()=>{
        const historyMock ={
            length:10,
            push:jest.fn(),
            goBack:jest.fn()
        }
        const wrapper = mount(
            <MemoryRouter initialEntries={['/hero/marvel-spider12332123']}>
                <Route path="/hero/:heroeId" component={()=><HeroScreen history={historyMock} />} />
            </MemoryRouter>
        );
        //expect(wrapper.find('Redirect').exists()).toBe(true);
        expect(wrapper.text()).toBe('');
    });


});