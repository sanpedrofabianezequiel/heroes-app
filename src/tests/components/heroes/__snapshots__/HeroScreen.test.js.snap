// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueabs en <HeroScreen/> debe de mostrar el componente redirect si no hay argumentos en el URL 1`] = `""`;
